ql/src/test/results/clientpositive/llap/sharedwork_semi.q.out

```diff
@@ -541,7 +541,7 @@ STAGE PLANS:
             Map Operator Tree:
                 TableScan
                   alias: s
-                  filterExpr: (ss_sold_date_sk is not null and ((ss_sold_date_sk BETWEEN DynamicValue(RS_7_d_d_date_sk_min) AND DynamicValue(RS_7_d_d_date_sk_max) and in_bloom_filter(ss_sold_date_sk, DynamicValue(RS_7_d_d_date_sk_bloom_filter))) or (ss_sold_date_sk BETWEEN DynamicValue(RS_21_d_d_date_sk_min) AND DynamicValue(RS_21_d_d_date_sk_max) and in_bloom_filter(ss_sold_date_sk, DynamicValue(RS_21_d_d_date_sk_bloom_filter))))) (type: boolean)
+                  filterExpr: (((ss_sold_date_sk BETWEEN DynamicValue(RS_7_d_d_date_sk_min) AND DynamicValue(RS_7_d_d_date_sk_max) and in_bloom_filter(ss_sold_date_sk, DynamicValue(RS_7_d_d_date_sk_bloom_filter))) or (ss_sold_date_sk BETWEEN DynamicValue(RS_21_d_d_date_sk_min) AND DynamicValue(RS_21_d_d_date_sk_max) and in_bloom_filter(ss_sold_date_sk, DynamicValue(RS_21_d_d_date_sk_bloom_filter)))) and ss_sold_date_sk is not null) (type: boolean)
```

Comment:
Iirc we order the expressions intentionally in such a way that the rest of expressions are evaluated before the SJ expression, since the probe of the bloom filter is usually more expensive than evaluating other expressions (heuristic).

Reply:
I've tried to retain the order - which have placed the bloom related checks at the end.

I recall that there was a ticket about ordering conditionals - but I can't find the related ticket; do I remember incorrectly?

Reply:
we see a case when NonBlockingOpDeDupProc merges FIL-FIL, the conditionals may be reorder. 
https://github.com/apache/hive/pull/1308